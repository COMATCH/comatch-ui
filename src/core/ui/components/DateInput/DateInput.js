import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import noop from 'lodash/noop';
import isNil from 'lodash/isNil';
import DatePicker from 'react-datepicker';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCalendarAlt } from '@fortawesome/free-regular-svg-icons/faCalendarAlt';
import moment from 'moment';
import { TextInput } from '../TextInput';

import './DateInput.scss';

/**
 * A convenience wrapper around the DatePicker component of the react-datepicker package.
 *
 * For a full list of all potential props check:
 * https://github.com/Hacker0x01/react-datepicker/blob/master/docs/datepicker.md
 */

function isValidDateValue(props, propName) {
    const prop = props[propName];

    // eslint-disable-next-line no-underscore-dangle
    if (!isNil(prop) && !prop._isAMomentObject && !(prop instanceof Date)) {
        return new Error(
            `Invalid prop \`${propName}\` supplied to` +
                ` DateInput. Expected a momentjs (or Date) instance, ` +
                `but got the value ${prop} of type ${typeof prop}. Validation failed.`,
        );
    }

    return null;
}

function getDateProp(date = null) {
    return date ? moment(date) : date;
}

const propTypes = {
    /**
     * A string according to which the date in the DateInput is formatted
     */
    dateFormat: PropTypes.string,
    inputError: PropTypes.string,
    label: PropTypes.string,
    /**
     * Needed by the DatePicker
     */
    locale: PropTypes.string.isRequired,
    maxDate: isValidDateValue,
    minDate: isValidDateValue,
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onClick: PropTypes.func,
    // eslint-disable-next-line valid-jsdoc
    /**
     * The value must be a moment.js object
     */
    value: isValidDateValue,
};

const defaultProps = {
    dateFormat: 'dd.mm.yyyy',
    inputError: '',
    label: '',
    maxDate: null,
    minDate: null,
    onClick: noop,
    value: moment(),
};

export const DateInput = ({
    locale,
    inputError,
    label,
    maxDate,
    minDate,
    name,
    onClick,
    onChange,
    value,
    ...props
}) => {
    if (isValidDateValue({ value }, 'value') instanceof Error) {
        return <span>ERROR! Invalid date value!</span>;
    }

    const [date, setDate] = useState(getDateProp(value));
    const popperClasses = classNames('DateInput__popper', { 'with-error': inputError });

    const handleChange = (newDate, event) => {
        setDate(moment(newDate));

        // in the onChange we pass value which would be either null or a date object
        onChange({
            ...event,
            target: {
                ...event.target,
                name,
                value: newDate,
            },
        });
    };

    // Component did update:
    useEffect(() => {
        setDate(value);
    }, [date, value]);

    const max = getDateProp(maxDate);
    const min = getDateProp(minDate);

    return (
        <DatePicker
            autoComplete="off"
            customInput={
                <TextInput
                    label={label}
                    inputError={inputError}
                    icon={<FontAwesomeIcon icon={faCalendarAlt} />}
                    name={name}
                    onChange={noop}
                    onClick={onClick}
                />
            }
            locale={locale}
            maxDate={max ? max.toDate() : max}
            minDate={min ? min.toDate() : min}
            name={name}
            nextMonthButtonLabel=""
            onChange={handleChange}
            popperClassName={popperClasses}
            previousMonthButtonLabel=""
            selected={date ? date.toDate() : date}
            utcOffset={0}
            {...props}
        />
    );
};

DateInput.propTypes = propTypes;
DateInput.defaultProps = defaultProps;
