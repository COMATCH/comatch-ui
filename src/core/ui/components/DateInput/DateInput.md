## DateInput

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**dateFormat** | `String` | `'dd.MM.yyyy'` | :x: | A string according to which the date in the DateInput is formatted NOTE: Keep in mind that `DD` & `YYYY` should be lowercase...
**inputError** | `String` | `''` | :x: | 
**label** | `String` | `''` | :x: | 
**locale** | `Enum('en', 'de', 'fr')` |  | :white_check_mark: | Needed by the DatePicker
**maxDate** | `isValidDateValue` | `null` | :x: | `Date` or `moment` or `null`
**minDate** | `isValidDateValue` | `null` | :x: | `Date` or `moment` or `null`
**name** | `String` |  | :white_check_mark: | 
**onChange** | `Function` |  | :white_check_mark: | 
**onClick** | `Function` | `noop` | :x: | 
**value** | `isValidDateValue` | `new Date()` | :x: | `Date` or `moment` or `null`

