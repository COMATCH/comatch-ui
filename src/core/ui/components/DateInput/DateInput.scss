@import '../../../styles/variables.scss';

$co-date-input-width: 245px;

.DateInput {
    &__popper {
        z-index: 1000000;

        &[data-placement='top-start'] {
            .react-datepicker__triangle {
                top: initial;
                bottom: 0;
                left: 20%;
                background-color: $co-color-white;
                transform: rotate(-135deg) translateX(-70%);
            }
        }

        .react-datepicker {
            box-shadow: 0 0 4px 0 rgba(119, 119, 119, 0.2);
            border: 1px solid $co-color-mid-gray;
            border-radius: 3px;
            width: $co-date-input-width;
            top: 15px;
            position: relative;

            &__triangle {
                background-color: $co-color-light-gray;
                border-left: 1px solid $co-color-mid-gray;
                border-top: 1px solid $co-color-mid-gray;
                width: 16px;
                height: 16px;
                position: absolute;
                left: 80%;
                top: 0;
                transform: rotate(45deg) translateX(-80%);
            }

            &__navigation {
                &--previous,
                &--next {
                    border: none;
                    background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxMCIgaGVpZ2h0PSIxNCIgdmlld0JveD0iMCAwIDEwIDE0Ij4gICAgPHBhdGggZmlsbD0iIzNFM0UzRSIgZmlsbC1ydWxlPSJub256ZXJvIiBkPSJNOS4yNDkgNy4zOGwtNi4yIDYuMjQyYS41MTIuNTEyIDAgMCAxLS4zNzYuMTYuNTEyLjUxMiAwIDAgMS0uMzc3LS4xNkwuOTEgMTIuMjI1YS41Mi41MiAwIDAgMS0uMTU4LS4zNzkuNTIuNTIgMCAwIDEgLjE1OC0uMzc5TDUuMzQ1IDcgLjkxIDIuNTMzYS41Mi41MiAwIDAgMS0uMTU4LS4zOC41Mi41MiAwIDAgMSAuMTU4LS4zNzhMMi4yOTYuMzc4YS41MTIuNTEyIDAgMCAxIC4zNzctLjE2Yy4xNDUgMCAuMjcuMDU0LjM3Ni4xNmw2LjIgNi4yNDNBLjUyLjUyIDAgMCAxIDkuNDA3IDdhLjUyLjUyIDAgMCAxLS4xNTguMzh6Ii8+PC9zdmc+');
                    background-color: transparent;
                    background-size: contain;
                    background-repeat: no-repeat;
                    cursor: pointer;
                    outline: 0;
                    position: absolute;
                    top: 18px;
                    height: 12px;
                }

                &--previous {
                    margin-left: 5px;
                    transform: rotate(180deg);
                }

                &--next {
                    right: 5px;
                }
            }

            &__header {
                background-color: $co-color-light-gray;
                border-bottom: 1px solid $co-color-mid-gray;

                .react-datepicker__day-name:first-child {
                    margin-left: 8px;
                }

                .react-datepicker__day-name:last-child {
                    margin-right: 8px;
                }
            }

            &__current-month {
                font-weight: bold;
                font-size: 14px;
                text-transform: uppercase;
                text-align: center;
                padding: 15px 0 5px 0;
            }

            &__day-names,
            &__month {
                width: $co-date-input-width;
                display: flex;
                flex-wrap: wrap;
            }

            &__month-container {
                overflow: hidden;
            }

            &__month {
                background-color: $co-color-white;
            }

            &__week {
                display: flex;
                width: $co-date-input-width;

                &:first-child {
                    .react-datepicker__day {
                        margin-top: 7.5px;
                    }
                }

                &:last-child {
                    .react-datepicker__day {
                        margin-bottom: 7.5px;
                    }
                }

                .react-datepicker__day:first-child {
                    margin-left: 8px;
                }

                .react-datepicker__day-name:last-child,
                .react-datepicker__day:last-child {
                    margin-right: 8px;
                }
            }

            &__day,
            &__day-name {
                font-size: 12px;
                width: 32.7px;
                text-align: center;
            }

            &__day {
                border-radius: 3px;
                transition: color 250ms ease-out;
                padding: 7.5px 0;
                position: relative;
                z-index: 1;

                &:after {
                    border-radius: 3px;
                    content: '';
                    display: block;
                    width: 22px;
                    height: 22px;
                    position: absolute;
                    top: 50%;
                    left: 50%;
                    transform: translate(-50%, -50%);
                    transition: background-color 250ms ease-out;
                    z-index: -1;
                }

                &--disabled {
                    cursor: initial;
                    color: $co-color-gray;

                    &:after {
                        background-color: $co-color-mid-gray;
                    }
                }

                &--selected,
                &--keyboard-selected {
                    color: $co-color-white;

                    &:after {
                        background-color: $co-color-primary;
                    }
                }

                &:hover {
                    &:not(.react-datepicker__day--disabled) {
                        cursor: pointer;
                    }

                    &:not(.react-datepicker__day--disabled):not(.react-datepicker__day--selected):not(.react-datepicker__day--keyboard-selected) {
                        &:after {
                            background-color: $co-color-light-gray;
                        }
                    }
                }
            }

            &__day-name {
                font-weight: bold;
                padding: 10px 0;
            }
        }
    }

    // States
    &__popper.with-error {
        .react-datepicker {
            top: -8px;
        }
    }
}
