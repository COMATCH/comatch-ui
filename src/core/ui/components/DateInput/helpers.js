import moment from 'moment';
import isNil from 'lodash/isNil';
import isNull from 'lodash/isNull';

/**
 * A reusable validator function. The function should be used in propTypes validation
 * but it's not limited to that use-case alone.
 *
 * @param {Object} props [Usually the component's props but could be any obj.]
 * @param {string} propName [The prop's name - e.g. `value`, `minDate`, `maxDate`, etc.]
 * @param {string} componentName [The component's name]
 * @returns {null|Error} []
 */
function isValidDateValue(props, propName, componentName = 'DateInput') {
    const prop = props[propName];

    // eslint-disable-next-line no-underscore-dangle
    if (!isNil(prop) && !prop._isAMomentObject && !(prop instanceof Date)) {
        return new Error(`
            Invalid prop \`${propName}\` supplied to ${componentName}. Expected a momentjs (or Date) instance,
            but instead got the value ${prop} of type ${typeof prop}. Validation failed.
        `);
    }

    return null;
}

/**
 * Guard method which will ensure that the underlying value will be compliant with
 * `react-datepicker`.
 *
 * @param {Date|moment|null} propValue [A date value which should be passed to `react-datepicker`]
 * @returns {Date|null} []
 */
function getSafeDateValue(propValue) {
    if (moment.isMoment(propValue)) {
        return propValue.toDate();
    }

    if (propValue instanceof Date || isNull(propValue)) {
        return propValue;
    }

    return null;
}

export { getSafeDateValue, isValidDateValue };
