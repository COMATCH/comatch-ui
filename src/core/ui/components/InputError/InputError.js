import React from 'react';
import PropTypes from 'prop-types';

import { StyledWrapper } from './StyledWrapper';

const propTypes = {
    /**
     * Text content of `InputError`
     */
    text: PropTypes.string.isRequired,
};

const defaultProps = {};

/**
 * An error indicator next to inputs for client side validation errors on form submissions.
 */

export const InputError = ({ text }) => (
    <StyledWrapper className="InputError">{text}</StyledWrapper>
);

InputError.defaultProps = defaultProps;

InputError.propTypes = propTypes;
