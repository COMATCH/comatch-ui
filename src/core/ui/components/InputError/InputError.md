
## InputError

From [`InputError.js`](InputError.js)

An error indicator next to inputs for client side validation errors on form submissions.

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**text** | `String` |  | :white_check_mark: | Text content of `InputError`



