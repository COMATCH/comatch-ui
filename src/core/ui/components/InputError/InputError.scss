@import '../../../styles/variables.scss';

// Explanatory error message
// for erroneous Input components.
.InputError {
    color: $co-color-red;
    display: block;
    font-size: $co-font-sm;
    margin: 7px 0 10px 0;
}
