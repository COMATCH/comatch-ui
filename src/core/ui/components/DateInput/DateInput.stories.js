/* eslint-disable import/no-unresolved */
import React from 'react';
import moment from 'moment';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { DateInput } from './DateInput';

const initialValue = moment().subtract(1, 'year');
const minDate = moment().subtract(5, 'days');
const maxDate = moment().add(5, 'days');

storiesOf('DateInput', module)
    .add('with basic style and English locale', () => (
        <>
            <div style={{ width: '250px' }}>
                <DateInput locale="en" name="field" onChange={action('changed')} />
            </div>
            <Highlight className="html">
                {`<DateInput locale="en" name="field" onChange={action('changed')} />`}
            </Highlight>
        </>
    ))
    .add('with label', () => (
        <>
            <div style={{ width: '250px' }}>
                <DateInput locale="en" label="Date input" name="field" onChange={action('changed')} />
            </div>
            <Highlight className="html">
                {`<DateInput locale="en" label="Date input" name="field" onChange={action('changed')} />`}
            </Highlight>
        </>
    ))
    .add('with label and disabled state', () => (
        <>
            <div style={{ width: '250px' }}>
                <DateInput locale="en" label="Date input" name="field" onChange={action('changed')} disabled />
            </div>
            <Highlight className="html">
                {`<DateInput locale="en" label="Date input" name="field" onChange={action('changed')} />`}
            </Highlight>
        </>
    ))
    .add('with label and initial value', () => (
        <>
            <div style={{ width: '250px' }}>
                <DateInput
                    locale="en"
                    label="Date input"
                    name="field"
                    value={initialValue}
                    onChange={action('changed')}
                />
            </div>
            <Highlight className="html">
                {`<DateInput locale="en" label="Date input"` +
                    ` name="field" value={initialValue} onChange={action('changed')} />`}
            </Highlight>
        </>
    ))
    .add('with label and error', () => (
        <>
            <div style={{ width: '250px' }}>
                <DateInput
                    locale="en"
                    label="Date input"
                    inputError="Broken date"
                    name="field"
                    onChange={action('changed')}
                />
            </div>
            <Highlight className="html">
                {`<DateInput locale="en" label="Date input"` +
                    ` inputError="Broken date" name="field" onChange={action('changed')} />`}
            </Highlight>
        </>
    ))
    .add('with min and max date', () => (
        <>
            <div style={{ width: '250px' }}>
                <DateInput
                    locale="en"
                    label="Date input"
                    minDate={minDate}
                    maxDate={maxDate}
                    name="field"
                    onChange={action('changed')}
                />
            </div>
            <Highlight className="html">
                {'<DateInput locale="en"' +
                    '\n\tlabel="Date input"' +
                    '\n\tminDate={minDate}' +
                    '\n\tmaxDate={maxDate}' +
                    '\n\tname="field"' +
                    "\n\tonChange={action('changed')} />"}
            </Highlight>
        </>
    ))
    .add('with German locale', () => (
        <>
            <div style={{ width: '250px' }}>
                <DateInput locale="de" name="field" onChange={action('changed')} />
            </div>
            <Highlight className="html">
                {`<DateInput locale="de" name="field" onChange={action('changed')} />`}
            </Highlight>
        </>
    ))
    .add('with French locale', () => (
        <>
            <div style={{ width: '250px' }}>
                <DateInput locale="fr" name="field" onChange={action('changed')} />
            </div>
            <Highlight className="html">
                {`<DateInput locale="fr" name="field" onChange={action('changed')} />`}
            </Highlight>
        </>
    ));
