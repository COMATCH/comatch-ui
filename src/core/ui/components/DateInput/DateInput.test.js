import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { lorem } from 'faker';
import moment from 'moment';
import noop from 'lodash/noop';
import '../../../config/tests/setup';

import { DateInput } from './DateInput';
import { InputLabel } from '../InputLabel';
import { InputError } from '../InputError';

describe('DateInput', () => {
    const REQUIRED_PROPS = {
        name: 'date-input-unit-test',
        locale: 'en',
        onChange: jest.fn(),
    };

    const OPTIONAL_PROPS = {
        dateFormat: 'yyyy.mm.dd',
        inputError: 'Testing InputError Message',
        label: 'Testing Label',
        value: moment('2019-01-01T10:00:00'),
        onClick: jest.fn(),
    };

    it('should render correctly', () => {
        const props = {
            ...REQUIRED_PROPS,
            value: moment('2019-02-08T10:00:00'),
        };

        const dateInputComponent = renderer.create(<DateInput {...props} />).toJSON();
        expect(dateInputComponent).toMatchSnapshot();
    });

    it('should render with default props', () => {
        const props = {
            ...REQUIRED_PROPS,
        };

        const dateInputComponent = mount(<DateInput {...props} />);

        // Required props:
        expect(dateInputComponent.prop('name')).toEqual(props.name);
        expect(dateInputComponent.prop('onChange')).toEqual(props.onChange);
        expect(dateInputComponent.prop('locale')).toEqual(props.locale);

        // Defaults:
        expect(dateInputComponent.prop('dateFormat')).toEqual('dd.mm.yyyy');
        expect(dateInputComponent.prop('inputError')).toEqual('');
        expect(dateInputComponent.prop('label')).toEqual('');
        expect(dateInputComponent.prop('onClick')).toEqual(noop);
        expect(dateInputComponent.prop('value') instanceof moment).toBeTruthy();
    });

    it('should render with custom optional props', () => {
        const props = {
            ...REQUIRED_PROPS,
            ...OPTIONAL_PROPS,
        };

        const dateInputComponent = mount(<DateInput {...props} />);

        // Required Props:
        expect(dateInputComponent.prop('name')).toEqual(props.name);
        expect(dateInputComponent.prop('onChange')).toEqual(props.onChange);

        // Optional Props:
        expect(dateInputComponent.prop('dateFormat')).toEqual(props.dateFormat);
        expect(dateInputComponent.prop('inputError')).toEqual(props.inputError);
        expect(dateInputComponent.prop('inputError')).toEqual(props.inputError);
        expect(dateInputComponent.prop('label')).toEqual(props.label);
        expect(dateInputComponent.prop('onClick')).toEqual(props.onClick);
    });

    it('should render correctly with a Label', () => {
        const inputWithLabel = (
            <DateInput {...REQUIRED_PROPS} value={moment('2019-02-08T10:00:00')} label={OPTIONAL_PROPS.label} />
        );
        const dateInputComponentWithLabel = renderer.create(inputWithLabel).toJSON();
        const mountedDateInput = mount(inputWithLabel);
        const dateInputContainerNode = mountedDateInput.children().at(0);

        expect(dateInputComponentWithLabel).toMatchSnapshot();
        expect(dateInputContainerNode.find(InputLabel).length).toBeGreaterThan(0);
    });

    it('should render correctly with an InputError', () => {
        const inputWithInputError = (
            <DateInput
                {...REQUIRED_PROPS}
                value={moment('2019-02-08T10:00:00')}
                inputError={OPTIONAL_PROPS.inputError}
            />
        );
        const mountedDateInput = mount(inputWithInputError);
        const dateInputContainerNode = mountedDateInput.children().at(0);
        const dateInputComponentWithInputError = renderer.create(inputWithInputError).toJSON();

        expect(dateInputComponentWithInputError).toMatchSnapshot();
        expect(dateInputContainerNode.find(InputError).length).toBeGreaterThan(0);
    });

    it('should throw a propType exception if value is truthy and not a moment or Date instance', () => {
        /* eslint-disable no-console */
        console.error = jest.fn();

        const props = {
            ...REQUIRED_PROPS,
        };

        mount(<DateInput {...props} value={lorem.word()} />);
        mount(<DateInput {...props} value={5} />);
        mount(<DateInput {...props} value={{}} />);

        expect(console.error).toHaveBeenCalledTimes(3);
        /* eslint-enable no-console */
    });

    it('should not throw a propType exception if a moment or Date instance or null is passed to value', () => {
        /* eslint-disable no-console */
        console.error = jest.fn();

        const props = {
            ...REQUIRED_PROPS,
        };

        mount(<DateInput {...props} value={moment()} />);
        mount(<DateInput {...props} value={new Date()} />);
        mount(<DateInput {...props} value={null} />);

        expect(console.error).toHaveBeenCalledTimes(0);
        /* eslint-enable no-console */
    });
});
