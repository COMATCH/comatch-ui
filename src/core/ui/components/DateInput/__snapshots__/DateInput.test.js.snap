// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput should render correctly 1`] = `
<div
  className="react-datepicker-wrapper"
>
  <div
    className="react-datepicker__input-container"
  >
    <div
      className="Input TextInput block with-icon"
      id={null}
    >
      
      <input
        autoComplete="off"
        className="TextInput__input"
        disabled={false}
        max={null}
        maxLength={null}
        min={null}
        name="date-input-unit-test"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        placeholder=""
        type="text"
        value="08.02.2019"
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-calendar-alt fa-w-14 "
        data-icon="calendar-alt"
        data-prefix="far"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48zm0 96v80h-96v-80h96zM176 352v-80h96v80h-96zm96 32v80h-96v-80h96zm-128-32H48v-80h96v80zm32-112v-80h96v80h-96zm128 32h96v80h-96v-80zM144 160v80H48v-80h96zM48 458v-74h96v80H54c-3.3 0-6-2.7-6-6zm346 6h-90v-80h96v74c0 3.3-2.7 6-6 6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      
    </div>
  </div>
</div>
`;

exports[`DateInput should render correctly with a Label 1`] = `
<div
  className="react-datepicker-wrapper"
>
  <div
    className="react-datepicker__input-container"
  >
    <div
      className="Input TextInput block with-icon"
      id={null}
    >
      <label
        className="InputLabel"
        htmlFor={null}
        id={null}
      >
        Testing Label
      </label>
      <input
        autoComplete="off"
        className="TextInput__input"
        disabled={false}
        max={null}
        maxLength={null}
        min={null}
        name="date-input-unit-test"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        placeholder=""
        type="text"
        value="08.02.2019"
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-calendar-alt fa-w-14 "
        data-icon="calendar-alt"
        data-prefix="far"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48zm0 96v80h-96v-80h96zM176 352v-80h96v80h-96zm96 32v80h-96v-80h96zm-128-32H48v-80h96v80zm32-112v-80h96v80h-96zm128 32h96v80h-96v-80zM144 160v80H48v-80h96zM48 458v-74h96v80H54c-3.3 0-6-2.7-6-6zm346 6h-90v-80h96v74c0 3.3-2.7 6-6 6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      
    </div>
  </div>
</div>
`;

exports[`DateInput should render correctly with an InputError 1`] = `
<div
  className="react-datepicker-wrapper"
>
  <div
    className="react-datepicker__input-container"
  >
    <div
      className="Input TextInput block has-error with-icon"
      id={null}
    >
      
      <input
        autoComplete="off"
        className="TextInput__input"
        disabled={false}
        max={null}
        maxLength={null}
        min={null}
        name="date-input-unit-test"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        placeholder=""
        type="text"
        value="08.02.2019"
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-calendar-alt fa-w-14 "
        data-icon="calendar-alt"
        data-prefix="far"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48zm0 96v80h-96v-80h96zM176 352v-80h96v80h-96zm96 32v80h-96v-80h96zm-128-32H48v-80h96v80zm32-112v-80h96v80h-96zm128 32h96v80h-96v-80zM144 160v80H48v-80h96zM48 458v-74h96v80H54c-3.3 0-6-2.7-6-6zm346 6h-90v-80h96v74c0 3.3-2.7 6-6 6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span
        className="InputError"
      >
        Testing InputError Message
      </span>
    </div>
  </div>
</div>
`;
